src/app-layout/__tests__/use-drawers.test.tsx

```diff
@@ -0,0 +1,100 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { act, render } from '@testing-library/react';
+import { TOOLS_DRAWER_ID, useDrawers } from '../../../lib/components/app-layout/utils/use-drawers';
+import { AppLayoutProps } from '../../../lib/components/app-layout';
+import { fireNonCancelableEvent } from '../../../lib/components/internal/events';
+import { testDrawer } from './utils';
+import { awsuiPlugins, awsuiPluginsInternal } from '../../../lib/components/internal/plugins/api';
+import { DrawerConfig } from '../../../lib/components/internal/plugins/controllers/drawers';
+
+const someDrawers = [
+  { ...testDrawer, id: 'one' },
+  { ...testDrawer, id: 'two' },
+];
+
+function delay() {
+  return act(() => new Promise(resolve => setTimeout(resolve)));
+}
+
+function TestComponent({
```

Comment:
Testing all corner cases of this hook via real app layouts is tedious, so I decided to start this separate unit tests for drawers handling only